### ISSUE TITLE
[$500] Android - Deeplink navigation does not work for RHN links.

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#27168]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Action Performed:
1. Open the app on Android.
2. Log out if already logined.
3. Use deeplink `new-expensify://settings/profile/personal-details/date-of-birth`. Please use the domain/scheme that works for you. you can also use staging url if testing staging app.
4. Now, login with a user.
5. Notice that user is not taken to deeplink page.

## Expected Result:
Deeplink for RHN should take the user to that page after login.

## Actual Result:
RHN deeplinks does not open after sign in when opened while user is log out.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.67.2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/fd077a6a-7b4b-4ecc-8df6-2391965c04bc

**Expensify/Expensify Issue URL:**
**Issue reported by:** @parasharrajat 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1694080489810799

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f1e03cd76963690d</li>
        <li>Upwork Job ID: 1701262065942163456</li>
        <li>Last Price Increase: 2024-03-18</li>
    </ul>
</details>

**Comment by sminrana** [comment]
-> Whenever you get the Intent, save you have a pending intent, in local storage. 
-> When the user log in, check if there is any pending intent, then take the user to Settings -> DOB view in this case.

**Comment by abekkala** [comment]
on hold

**Comment by abekkala** [comment]
Not overdue. On Hold

**Comment by abekkala** [comment]
@hungvu193 
https://github.com/Expensify/App/issues/26021#issuecomment-1714222456

https://github.com/Expensify/App/issues/26021#issuecomment-1719003621

**Comment by abekkala** [comment]
@lukemorawski can you link the PR here

**Comment by lukemorawski** [comment]
@abekkala I'm working on that today. Will post a PR today or tomorrow!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by narefyev91** [comment]
@lukemorawski hey! can you please re-test on latest main if issue is still reproduced on latest main without your changes

**Comment by yuwenmemon** [comment]
I'm heading OOO on sabbatical so going to assign this one out to another internal engineer.

**Comment by abekkala** [comment]
waiting on PR

**Comment by abekkala** [comment]
@narefyev91 can you complete the checklist above?

**Comment by narefyev91** [comment]
- The PR that introduced the bug has been identified. Link to the PR - it was not a regression from any PR just issue of unhandled deep links.
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment - N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion - N/A
- Determine if we should create a regression test for this bug - N/A

**Comment by abekkala** [comment]
### PAYMENT DUE: 
Issue reported by: @parasharrajat [$50]  (🚫 Upwork payment, please manually request in Expensify Chat)

Fix: @lukemorawski does not require payment (Contractor)
PR Review: @narefyev91 does not require payment (Contractor)

**Comment by abekkala** [comment]
No, I don't see that we need further proposals on this one. The PR for this one has been in production for 2 weeks. 

This one is done.

**Comment by abekkala** [comment]
### LAST ACTION NEEDED: 

@parasharrajat [$50] (🚫 Upwork payment, please manually request in Expensify Chat)

@narefyev91 please complete checklist

**Comment by thienlnam** [comment]
Okay it looks like we're still working on this one. The original [PR](https://github.com/Expensify/App/pull/28278) was reverted.

@lukemorawski Are you still able to work on this, updating your PR to test for this [regression](https://github.com/Expensify/App/issues/29642) as well?

**Comment by lukemorawski** [comment]
@thienlnam sure, will have a go tomorrow!

**Comment by thienlnam** [comment]
We're still working on this - @lukemorawski is planning on getting another PR up

**Comment by abekkala** [comment]
Hi @lukemorawski do you have a PR for this?

**Comment by lukemorawski** [comment]
@abekkala working on that now

**Comment by abekkala** [comment]
I'm heading out for the Thanksgiving holiday and will be back bright and early Monday morning.
If something pops up that needs urgent attention please reapply label or ask in slack.

**Comment by abekkala** [comment]
@thienlnam I want to make sure I'm understanding this one correctly....we're waiting for another fix due to regression that this PR caused, correct?

**Comment by thienlnam** [comment]
Yup - that's correct. @lukemorawski when do you think you can get back to this? If you're working on higher priority stuff we can try to get this re-assigned

**Comment by lukemorawski** [comment]
@thienlnam I could take a look today, but from tomorrow (22.12) I'm OOO till 2.01.24.

**Comment by thienlnam** [comment]
Yeah no worries, most of us will be out the next couple of weeks - if you could get back on this after the break that would be great

**Comment by lukemorawski** [comment]
@thienlnam I think I got it solved and by solved I mean the problems mentioned here, but also [here](https://github.com/Expensify/App/pull/31316) and others. Should I prepare a PR for this ticket?

**Comment by thienlnam** [comment]
Oh, if the problem has been solved in another PR we're good to close. But the original PR https://github.com/Expensify/App/pull/31316 was reverted due to the other issues it caused - are you saying this issue is no longer reproducible?

**Comment by lukemorawski** [comment]
@thienlnam Sorry, maybe I've made my self no clear enough. I have found a solution to following issues that seem to be caused by the PR you mentioned:
https://github.com/Expensify/App/issues/33205
https://github.com/Expensify/App/issues/33001
Should I post a PR to this ticket and just mention those listed issues?

**Comment by thienlnam** [comment]
Oh yes, please do that!

**Comment by thienlnam** [comment]
Oh I see, though it looks like both of those issues already have a solution in progress - so if you just re-add the changes in https://github.com/Expensify/App/pull/31316 after those PRs are merged would that fix this current issue?

**Comment by lukemorawski** [comment]
@thienlnam yes it would fix that

**Comment by thienlnam** [comment]
Yeah, let's go forward with that then

**Comment by mallenexpensify** [comment]
Removed `Reviewing` and bumped to `Weekly` so we can get regular updates. Partially because the below is held on this. 
- https://github.com/Expensify/App/issues/23899#issuecomment-1953409599 

@abekkala what wave should this be a part of?  #vip-vsb ?  Seem like it.

**Comment by abekkala** [comment]
@mallenexpensify ah ok, I probably misunderstood the [comment above](https://github.com/Expensify/App/issues/27168#issuecomment-1875946186). I was thinking that @thienlnam comment insinuated that once the PR merged for the fix for https://github.com/Expensify/App/issues/33205 and https://github.com/Expensify/App/issues/33001 it could be used for https://github.com/Expensify/App/pull/31316 which would _then_ resolve this one. 

I was thinking after PR merge I'd just re-test weekly on this one to see if it needed to proceed

**Comment by mallenexpensify** [comment]
Thanks @abekkala , it looks like all the above are closed/merged.  If you want Applause to test, I'd either add the `retest-weekly` label or drop in #qa and ask them to test (and tag folks bookmarked at top).

**Comment by abekkala** [comment]
@kbecciv can you give me a status on the retest?

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests with a slightly different behavior 


https://github.com/Expensify/App/assets/43995119/20635e01-29a5-4bc8-a5fe-4e22282ffe8f

**Comment by abekkala** [comment]
@mvtglobally If there is now different behavior from the OP
can you provide a new:

- Action Performed:
- Expected Result:
- Actual Result:

**Comment by abekkala** [comment]
@shubham1206agra the retest weekly label was applied to see if the reported issue was still occurring. If you're not able to reproduce then we can close.

**Comment by mallenexpensify** [comment]
Thanks @shubham1206agra for reviewing some of these _stuck_ issues and helping to get some closed and.. hopefully fixed!

**Comment by mallenexpensify** [comment]
@parasharrajat , what's the reasoning for paying the reporting bonus here?

**Comment by abekkala** [comment]
@mallenexpensify wouldn't a reporting bonus only be paid for an issue that actually implemented a fix?

**Comment by mallenexpensify** [comment]
Yeah, that's what I was assuming too @abekkala, based off[ this in CONTRIBUTING.md](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#:~:text=Payment%20for%20your%20contributions%20will%20be%20made%20no%20less%20than%207%20days%20after%20the%20pull%20request%20is%20deployed%20to%20production)
> Payment for your contributions will be made no less than 7 days after the pull request is deployed to production

I know CONTRIBUTING.md has been updated to remove details on the reporting bounty but my memory is that we always stated the bug had to be fixed in order to be paid.

**Comment by abekkala** [comment]
This one does not qualify for a reporting payout of $50 given the report did not result in a PR being created and deployed for a fix
